CDCgov/prime-reportstream
# Add additional reporting output file options

This PR allows us to create two other types of output files for our schemas, specifically in a data dictionary format so we can create a CSV or Excel XLSX file. The data dictionary file format is based on the format we are using on the monkeypox effort to communicate our schema structure to other teams in the CDC.

This also includes some changes and cleanup of the documentation code to collect common functionality into base classes.

Test Steps:
1. Build the application
2. Run `./prime generate-docs --output-dir build/xl --generate-xl --input-schema covid-19 --no-generate-markup` to generate an Excel document
3. Look in the `build/xl/` folder to locate the XSLX file and verify it's correct
4. Run `./prime generate-docs --output-dir build/csv --generate-csv --input-schema covid-19 --no-generate-markup` to generate a CSV document
5. Look in the `build/csv/` folder to location the CSV file and verify it's correct
6. Run unit tests
7. Run integration tests
8. Run `./prime test` 

## Changes
- Unifies logic into base classes so things are more clean in the documentation code
- Adds a CSV documentation factory generator
- Adds an Excel documentation factory generator
- Adds additional tests for some of the documentation logic
- Adds command line logic to include writing out CSV and Excel files

## Checklist

### Testing

### Process
